### PR TITLE
Change `BindingsName` to hold `Cow<str>` rather than  `String` to reduce `clone` needed for lookups

### PR DESCRIPTION
**Pull request by jpschorr**
---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by am357 — APPROVED** [review]

File: partiql-eval/src/eval/expr/path.rs [hunk 1]
```diff
@@ -19,58 +19,107 @@ pub(crate) struct EvalPath {
...
+        ctx: &dyn EvalContext,
+    ) -> Option<&'a Value> {
+        #[inline]
+        fn as_str<'v>(v: &'v Cow<Value>) -> Option<&'v str> {
+            match v {
+                Cow::Borrowed(Value::String(s)) => Some(s),
+                Cow::Owned(Value::String(s)) => Some(s.as_ref()),
+                _ => None,
+            }
+        }
```
Comment (alancai98) [inline partiql-eval/src/eval/expr/path.rs:43]: This fn is repeated in `take_val` L74-80. Are both needed?

**Review by alancai98 — COMMENTED** [review]
Looks good. Left a couple minor comments about things that could be deleted.

Looks like the CI fails because of some tests not using `Cow<str>` for the `BindingsName`. And the conformance test regression probably needs a rebase from `main` or the target branch's changes.

File: partiql-eval/src/eval/expr/path.rs [hunk 2]
```diff
@@ -19,58 +19,93 @@ pub(crate) struct EvalPath {
...
-                EvalPathComponent::KeyExpr(ke) => {
-                    let key = ke.evaluate(bindings, ctx);
-                    match (value, key.as_ref()) {
-                        (Value::Tuple(tuple), Value::String(key)) => {
-                            tuple.get(&BindingsName::CaseInsensitive(key.as_ref().clone()))
-                        }
-                        _ => None,
-                    }
+#[inline]
+fn as_str<'v>(v: &'v Cow<Value>) -> Option<&'v str> {
```
Comment (alancai98) [inline partiql-eval/src/eval/expr/path.rs:29]: Clippy warning for this fn

```
using a reference to `Cow` is not recommended

warning: using a reference to `Cow` is not recommended
  --> partiql-eval/src/eval/expr/path.rs:29:18
   |
29 | fn as_str<'v>(v: &'v Cow<Value>) -> Option<&'v str> {
   |                  ^^^^^^^^^^^^^^ help: change this to: `&Value`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg
   = note: `#[warn(clippy::ptr_arg)]` on by default
```

**Review by alancai98 — APPROVED** [review]
Looks good. Left a comment about a clippy warning emitted from CI

**Pull request merged**